Change cli interface

nomaddo opened the pull request:
This patch modify the CLI `VC4C` behaviors: allow users to omit specifying an output file when the number of input files are just one.
This behavior is natural compared with other compilers (like `gcc` with `-S` or `-c`).


### before

```
$ VC4C --bin -o ./test.cl.bin test.cl
```

### after

```
$ VC4C --bin test.cl
```

doe300 commented on src/main.cpp:
```diff
@@ -150,31 +150,17 @@ int main(int argc, char** argv)
     bool runDisassembler = false;
 
-    if(argc < 3)
+    int i = 1;
```
Why is the i not local to the loop?

nomaddo replied:
Just a bug. fixed.

nomaddo commented on src/main.cpp:
```diff
@@ -199,29 +190,29 @@ int main(int argc, char** argv)
+
             outputFile = argv[i + 1];
-            // any further parameter is an input-file
-            i += 2;
```
`i += 2` was just bug, fixed.

nomaddo commented:
> Can you please also rewrite the output-file block (check for -o) so it does not abort the loop, but simply continues it? This will allow specifying options after the output file.

Fixed.

> Can you please re-add that if the compiler is run without any arguments (argc == 1), the help is printed?

Fixed. Actually, the behavior you said is kinder....
